testing/test-setup/src/lib/extend/markdown-table.matcher.ts

```diff
@@ -0,0 +1,44 @@
+import type { SyncExpectationResult } from '@vitest/expect';
+import { expect } from 'vitest';
+
+export type CustomMarkdownTableMatchers = {
+  toContainMarkdownTableRow: (cells: string[]) => void;
+};
+
+expect.extend({
+  toContainMarkdownTableRow: assertMarkdownTableRow,
+});
+
+function assertMarkdownTableRow(
+  actual: string,
+  expected: string[],
+): SyncExpectationResult {
+  const rows = actual
+    .split('\n')
+    .map(line => line.trim())
+    .filter(line => line.startsWith('|') && line.endsWith('|'))
+    .map(line =>
+      line
+        .split('|')
+        .map(cell => cell.trim())
+        .filter(Boolean),
+    )
+    .filter(row => row.some(cell => /\w/.test(cell)));
```

Comment:
Not sure if these filters are a good idea. Empty cells or cells with non-word characters (e.g. `-`) are perfectly, so I don't see why we'd treat differently to other cells.

Reply:
The `.filter(Boolean)` is still present, would ignore empty cells.